### PR TITLE
tgui recompile comment bot

### PR DESCRIPTION
**Pull request by ZeWaka**
## About the PR
adds `/rebuild tgui` or `/rebuild-tgui` triggerable by the PR author, developer, or any triage member

will resolve tgui merge conflicts and rebuild

main rebuild task mainly by @MCterra10, adapted a bit for goonstation and trimmed down

## Why's this needed? 

people have been annoying me for like 9 months about this

attn: @ike709 @mordent-goonstation https://github.com/BeeStation/BeeStation-Hornet/pull/4738

### CODE REVIEW

**Review by mordent-goonstation — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
Does this work?

**Comment by ike709** [comment]
It sure doesn't on Bee.

File: .github/workflows/tgui_recompile.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      # Uses a smart little git hack to make a merge commit for just a limited set of files.
+      # Changed tgui/public to browserassets/tgui for goonstation.
+      - name: Conflict Resolution
+        if: ${{ github.event.client_payload.mergeable == false }}
+        run: |
+          git remote add base https://github.com/${{ github.event.client_payload.pr_head_full_repo_name }}.git
+          git fetch base --depth=25 ${{ env.DEFAULT_BRANCH }}
+          git merge -s ours --no-commit base/${{ env.DEFAULT_BRANCH }}
+          git checkout base/${{ env.DEFAULT_BRANCH }} browserassets/tgui
+          git commit -m "TGUI Reset" -a || true
```
Comment (Crossedfall) [inline .github/workflows/tgui_recompile.yml:47]: Why are you using the fork's master branch to resolve conflicts? I know of maybe three people that keep their master in-line with upstream

**Comment by MCterra10** [comment]
Zewaka you funny goonman, just a heads up that you can't remove the auto rebuild part of the action without causing your master branch builds to desync from the tgui changes in place, since conflict resolution doesn't do a full merge, only updates the tgui bundles for the changes on the pr

**Comment by ZeWaka** [comment]
> Zewaka you funny goonman, just a heads up that you can't remove the auto rebuild part of the action without causing your master branch builds to desync from the tgui changes in place, since conflict resolution doesn't do a full merge, only updates the tgui bundles for the changes on the pr 

That doesn't make sense to me. I see zero reason why it's needed.

**Pull request merged**